androidapp/app/src/main/java/com/example/lanweather/ui/home/HomeFragment.kt

```diff
@@ -16,12 +25,127 @@ class HomeFragment : Fragment() {
     }
 
     private lateinit var viewModel: HomeViewModel
+//    private val mainActivity: MainActivity = MainActivity.getInstance()
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return inflater.inflate(R.layout.home_fragment, container, false)
+        viewModel =
+            ViewModelProviders.of(this).get(HomeViewModel::class.java)
+
+        val root = inflater.inflate(R.layout.home_fragment, container, false)
+
+        val database = AppDatabase(MainActivity.getInstance().baseContext)
+
+        CoroutineScope(CoroutineName("PopulateHome")).launch {
+            withContext(Dispatchers.IO) {
+                // Incorporated this delay to allow the coroutine in the MainActivity to finish
```

Comment:
if something needs to finish shouldn't it be called blocking? any sort of sleep/delay in code is concerning, wondering if there's another way.

Reply:
I have a rudimentary understanding of Kotlin's coroutines. I was attempting to figure that out, but I decided to go with the sleep/delay option so that we could at least merge something that works and doesn't crash.

This coroutine isn't supposed to run until the coroutine in the MainActivity finishes.

Reply:
this might be helpful:
[https://stackoverflow.com/questions/53665470/wait-until-kotlin-coroutine-finishes-in-oncreateview](https://stackoverflow.com/questions/53665470/wait-until-kotlin-coroutine-finishes-in-oncreateview)

there also might be a way to `join()` the job from the `MainActivity` if this class has knowledge of the `MainActivity` and the job launched from the `MainActivity` gets assigned to a member variable.